### PR TITLE
fix: Invalid "seedfrom" in NoCloud system configuration

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
Use individual commits
```

## Additional Context
First commit is just pytestifying the tests so I can later use the caplog fixture.

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/unittests/sources/test_nocloud.py [hunk 1]
```diff
@@ -3,36 +3,31 @@
...
-class TestNoCloudDataSource(CiTestCase):
-    def setUp(self):
-        super(TestNoCloudDataSource, self).setUp()
-        self.tmp = self.tmp_dir()
-        self.paths = helpers.Paths(
-            {"cloud_dir": self.tmp, "run_dir": self.tmp}
-        )
+@pytest.fixture
+def paths(tmpdir):
+    yield helpers.Paths({"cloud_dir": tmpdir, "run_dir": tmpdir})
```
Comment (blackboxsw) [inline tests/unittests/sources/test_nocloud.py:20]: We already have a paths fixture in the [top-level conftest.py](https://github.com/canonical/cloud-init/blob/main/conftest.py#L201) so no need to reinvent here.

**Review by blackboxsw — APPROVED** [review]
thanks for the additional test coverage here asserting that sys_config files can still provide NoCloud.seedfrom urls.

**Comment by TheRealFalcon** [comment]
+1 to your change

**Comment by merkys7** [comment]
Hello guys! @TheRealFalcon @blackboxsw 
When will we see this fix on official releases?
I am having issues with the latest OS VMs and error:
```
2024-10-28 14:00:43,257 - util.py[WARNING]: Getting data from <class 'cloudinit.sources.DataSourceNoCloud.DataSourceNoCloudNet'> failed
```
I have tracked down that this error is fixed after your PR

**Comment by TheRealFalcon** [comment]
The next planned upstream release is for November 18th. The changes should hit stable Ubuntu releases 1-2 weeks after that.

**Pull request merged**